aapcs64/aapcs64.rst

```diff
@@ -3111,6 +3112,65 @@ instead.
 The SVE tuple types are mangled using their ``arm_sve.h`` names
 (``svBASExN_t``).
 
+Function types which have SME streaming, ZA interface or ZT0 interface attributes must include these attributes in the name mangling of the type.
+
+SME attributes are mangled in the same way as a template:
+
+  template<typename, unsigned, unsigned, unsigned> __SME_ATTRS;
+
+with the arguments:
+
+  __SME_ATTRS<normal_function_type, streaming_mode, za_state, zt0_state>;
+
+where:
+
+* normal_function_type is the function type without any SME attributes.
+
+* streaming_mode is an integer representing the streaming-mode of the function:
+
+  +------------------------+--------------------------+
+  | Interface Type         | Value                    |
+  +========================+==========================+
+  | Normal (default)       | 0                        |
+  +------------------------+--------------------------+
+  | Streaming Mode         | 1                        |
+  +------------------------+--------------------------+
+  | Streaming-Compatible   | 2                        |
+  +------------------------+--------------------------+
+
+* za_state is an integer representing the ZA state of the function:
+
+  +------------------------+--------------------------+
+  | Interface Type         | Value                    |
+  +========================+==========================+
+  | No ZA State (default)  | 0                        |
+  +------------------------+--------------------------+
+  | Shared ZA              | 1                        |
```

Comment:
The reason for mangling the ZA information is to deal with the fact that the sharing attributes create distinct function types for overloading purposes.  I think this means that we need to represent all five possibilities:

* normal
* `__arm_in("za")`
* `__arm_inout("za")`
* `__arm_out("za")`
* `__arm_preserves("za")`

Similarly for `zt0`.

Since that classification is only described in the ACLE, we should probably describe the mangling there too.  Sorry for only realising that now.

Reply:
Thanks @rsandifo-arm, I've updated this to represent each of the shared ZA/ZT0 attributes separately and moved these changes to the ACLE (https://github.com/ARM-software/acle/pull/358).